### PR TITLE
fpx_weight_only quantization

### PR DESCRIPTION
**Pull request by a-r-r-o-w**
The benchmarks are currently running. Will update here when complete

### CODE REVIEW

File: inference/benchmark_video.py [hunk 1]
```diff
@@ -156,7 +158,7 @@ def get_args():
         "--model_id",
         type=str,
         default="THUDM/CogVideoX-5b",
-        choices=["THUDM/CogVideoX-2b", "THUDM/CogVideoX-5b"],
+        # choices=["THUDM/CogVideoX-2b", "THUDM/CogVideoX-5b"],
```
Comment (sayakpaul) [inline inference/benchmark_video.py:161]: Why?

**Comment by sayakpaul** [comment]
Would be cool to propagate the changes to `benchmark_image.py` once the benchmarks are done.

File: inference/benchmark_video.py (hunk 1 above)
Comment (a-r-r-o-w) [reply inference/benchmark_video.py:161]: I'll undo this later. I'm using a local checkpoint path

Comment (a-r-r-o-w) [inline inference/benchmark_video.py:161]:
```suggestion
        choices=["THUDM/CogVideoX-2b", "THUDM/CogVideoX-5b"],
```

File: README.md [hunk 2]
```diff
@@ -293,6 +295,17 @@ With the newly added `fp8dqrow` scheme, the inference latency is **76.70 seconds
...
+|  model_type  |  compile  |  fuse_qkv  |  quantize_vae  |  quantization  |   model_memory |   inference_memory |   time  |
+|:------------:|:---------:|:----------:|:--------------:|:--------------:|:--------------:|:------------------:|:-------:|
+|      5B      |   False   |   False    |     False      |    fp6_e3m2    |          7.798 |             21.028 | 287.842 |
+|      5B      |   False   |    True    |     False      |    fp6_e3m2    |           8.63 |             23.243 | 285.294 |
+|      5B      |   False   |   False    |     False      |    fp5_e2m2    |          6.619 |              21.02 | 305.401 |
+|      5B      |   False   |    True    |     False      |    fp5_e2m2    |          7.312 |             23.237 | 304.725 |
+|      5B      |   False   |   False    |     False      |    fp4_e2m1    |          5.423 |             21.012 | 282.835 |
+|      5B      |   False   |    True    |     False      |    fp4_e2m1    |          5.978 |             23.228 | 280.262 |
+
 </details>
```
Comment (sayakpaul) [inline README.md:309]: Let's also include any interesting observations (like low latency and memory and good quality) if any.

File: README.md [hunk 3]
```diff
@@ -281,8 +281,10 @@ Through visual inspection of various outputs, we identified that the best result
 
 With the newly added `fp8dqrow` scheme, the inference latency is **76.70 seconds** for CogVideoX-5b (batch size: 1 , steps: 50, frames: 49, resolution: 720x480) on an H100. `fp8dqrow` has more scales per tensors and less quantization error. The quality, from visual inspection, is very close to fp16/bf16 and better than int8 in many cases.
 
+TorchAO also supports arbitary exponent and mantissa bits for floating point types, which provides experimental freedom to find the best settings for your models. Here, we also share results with `fp6_e3m2`, `fp5_e2m2` and `fp4_e2m1` that do well in preserving quality.
```
Comment (sayakpaul) [inline README.md:284]: Let's also include the note that it doesn't work with `torch.compile()` yet.

**Review by sayakpaul — APPROVED** [review]
Just two minor comments. 

I will post the image benchmarks in a separate PR and will open an issue for the AO team to look into.

**Pull request merged**